magento/pwa-studio
# [bugfix] Hide stale suggestions

## Description
Fix a bug with `Suggestions` wherein stale results would continue to be displayed when the field had been cleared. The proper fix would entail resetting the query, such that `called` would be `false` and `data` would be `undefined`, but Apollo does not yet offer that functionality. Instead, this fix recognizes when the field has been cleared and incorporates that into the conditional rendering logic of `Autocomplete` and `Suggestions`.

This fix involves changes to the talon APIs, so it's probably a major. As a result, I've also taken the opportunity to have `SearchBar` obtain `history` and `location` from hook calls rather than render props.

## Related Issue
[PWA-361](https://jira.corp.magento.com/browse/PWA-361)

## Acceptance 

### Verification Stakeholders
@dpatil-magento 

### Specification

### Verification Steps
1. Enter `athena` in the search input.
2. Verify that suggested categories and products appear.
3. Click the button to clear the search input.
4. Verify that suggestions are no longer displayed, but a prompt is.

## Screenshots / Screen Captures (if appropriate)

## Checklist
* I have updated the documentation accordingly, if necessary.
* I have not yet added tests to cover my changes, if necessary.
